app/models/vm_scan.rb

```diff
@@ -491,22 +492,26 @@ def call_abort_retry(*args)
     end
   end
 
+  def process_delete_evm_snapshot(vm)
```

Comment:
It seems this method is only necessary because of the differences between `vm_delete_evm_snapshot` methods of different providers. Ultimately, I think there should be a `delete_evm_snapshot` method in the VM object. Then each type of VM can have their own implementation with a common signature. Then all we'd have to do is call `vm.delete_evm_snapshot(mor)` without all the `if vm.kind_of?` checks.

For now, I think this fix is ok the way it is, but we should consider cleaning it up in the future.